crates/bevy_ecs/src/event/registry.rs

```diff
@@ -77,4 +77,26 @@ impl EventRegistry {
             }
         }
     }
+
+    /// Removes an event from the registry and world
+    pub fn deregister_events<T: Event>(world: &mut World) {
```

Comment:
Word of warning of how this will affect existing `EventReader`s
We can do this in another PR, but it's a pretty important side effect

Reply:
Let me clarify, if we remove the `Events<E>`, running a system with `EventReader` will obviously crash,
the question is what if we re-add the same `Events<E>` back, how will that affect the reader

Reply:
I think I'll adjust it so this doesn't remove `Events<E>` - ATM it'll cause the count to re-set, but since the readers aren't re-set (and I'm not sure if there's a way to remove or reset all `Local<'s, ManualEventReader<E>>` resources) re-introducing the Events<E> will lead to it missing all events until the count reaches the latest count a given reader saw.

Reply:
Actually - changed my mind here.

Since you'd have to remove the systems anyway to prevent crashes, the Local is unlikely to be retained. I do have a potential way to prevent both the crashes & this issue in mind (generational indexes), but think that might be a bit beyond the scope here.

Reply:
You can have run conditions or schedules that you don't run,
anyways, the solution is outside of the scope for this PR, no need to sweat it rn